packages/compass-indexes/src/components/indexes-toolbar/indexes-toolbar.tsx

```diff
@@ -36,30 +36,30 @@ type IndexesToolbarProps = {
   errorMessage: string | null;
   isReadonly: boolean;
   isReadonlyView: boolean;
-  preferencesReadOnly: boolean;
   isWritable: boolean;
   localAppRegistry: AppRegistry;
   isRefreshing: boolean;
   writeStateDescription?: string;
   onRefreshIndexes: () => void;
+  readOnly?: boolean;
 };
 
 export const IndexesToolbar: React.FunctionComponent<IndexesToolbarProps> = ({
   errorMessage,
   isReadonly,
   isReadonlyView,
-  preferencesReadOnly,
   isWritable,
   localAppRegistry,
   isRefreshing,
   writeStateDescription,
   onRefreshIndexes,
+  readOnly, // preferences readOnly.
 }) => {
   const onClickCreateIndex = useCallback(() => {
     localAppRegistry.emit('toggle-create-index-modal', true);
   }, [localAppRegistry]);
   const showCreateIndexButton =
-    !isReadonly && !isReadonlyView && !preferencesReadOnly && !errorMessage;
+    !isReadonly && !isReadonlyView && !readOnly && !errorMessage;
```

Comment:
Totally non-blocking, but if you’d rather not pass this as a prop (and you might want to do it for easier testing or similar), you can do 

```js
  const preferencesReadOnly = usePreference('readOnly', React);
  const showCreateIndexButton = 
    !isReadonly && !isReadonlyView && !preferencesReadOnly && !errorMessage;
```

(because I totally see that it’s confusing to have `isReadonly` and `readOnly` as prop names)

Reply:
Yeah, I found it handy for unit testing. But you would still replace it with usePreference? How then to tweak readOnly for testing?

Reply:
I decided to keep it as it is for now to get this PR merged, and revisit later as a separate chunk of work together with refactoring of `compass-aggregations` and `compass-crud` stores that still keep using `preferencesReadOnly` and listen for readOnly changed events.

Reply:
The follow up ticket: https://jira.mongodb.org/browse/COMPASS-6300